### ISSUE TITLE
Internationalization: Read localized strings from extension JSON

### ISSUE THREAD

**Issue by tjcouch-sil** [paranext/paranext-core#740]
As an extension developer, I want to be able to provide localized string files so my extension can be localized for users.

----

Depends on #419, #739

On the localized string service:
- [ ] Maintain list of all localized strings and their metadata provided by extensions in their manifest files and built into core instead of the list just being loaded only from core files
- [ ] If #742 is finished, move localized strings from core to their associated extensions where appropriate.

**Comment by tjcouch-sil** [comment]
@katherinejensen00 Yep! Oops. Closed as finished with #739

**Comment by roopa0222** [comment]
Test Repo steps,

In my test extension 
In my web-vew tsx I used the PAPI useSetting hook to set the locale language
```
const [language, setLanguage] = useSetting('platform.interfaceLanguage', ['en']);

```
I updated the localizedStrings.json with the following
```
{
  "metadata": {},
  "localizedStrings": {
    "en": {
      "%mainMenu_openPAPIMenuTestScreen%": "Open PAPI Menu TestScreen",
      "%webView_paranextExtTesting_results%": "Test Results",
      "%webView_paranextExtTesting_results_pass%": "Pass",
      "%webView_paranextExtTesting_results_fail%": "Fail"
    },
    "te": {
      "%webView_paranextExtTesting_results_pass_te%": "పాస్",
      "%webView_paranextExtTesting_results_fail_te%": "విఫలం"
    }
  }
}
```
In my menus.json I have the following .
```
 "webViewMenus": {
    "paranextExtTesting.reactTC18": {
      "includeDefaults": true,
      "topMenu": {
        "columns": {
          "paranextExtTesting.results": {
            "label": "%webView_paranextExtTesting_results%",
            "order": 4
          }
        },
        "groups": {
          "paranextExtTesting.resultsTop": {
            "column": "paranextExtTesting.results",
            "order": 1
          }
        },
        "items": [
          {
            "label": "%webView_paranextExtTesting_results_pass%",
            "group": "paranextExtTesting.resultsTop",
            "order": 1,
            "command": "paratextExtensionTest.resultsPass"
          },
          {
            "label": "%webView_paranextExtTesting_results_fail%",
            "group": "paranextExtTesting.resultsTop",
            "order": 2,
            "command": "paratextExtensionTest.resultsFail"
          }
        ]
      },
      "contextMenu": {
        "groups": {},
        "items": []
      }
    },
    "paranextExtTesting.react2": {
      "includeDefaults": false,
      "topMenu": {
        "columns": {
          "paranextExtTesting.results": {
            "label": "%webView_paranextExtTesting_results%",
            "order": 1
          }
        },
        "groups": {
          "paranextExtTesting.resultsTop": {
            "column": "paranextExtTesting.results",
            "order": 1
          }
        },
        "items": [
          {
            "label": "%webView_paranextExtTesting_results_pass_te%",
            "group": "paranextExtTesting.resultsTop",
            "order": 1,
            "command": "papiMenuTest.resultsPass"
          },
          {
            "label": "%webView_paranextExtTesting_results_fail_te%",
            "group": "paranextExtTesting.resultsTop",
            "order": 2,
            "command": "papiMenuTest.resultsFail"
          }
        ]
      },
      "contextMenu": {
        "groups": {},
        "items": []
      }
    }
  }
```
Working as expected.

![Screenshot 2024-05-16 102354](https://github.com/paranext/paranext-core/assets/133702629/6c0bc105-741c-4722-b17c-5e34a82c8901)
![Screenshot 2024-05-16 102322](https://github.com/paranext/paranext-core/assets/133702629/a8397634-f382-4b39-964f-df3f7e966736)
